docs/adr/0002-database-corruption.md

```diff
@@ -0,0 +1,132 @@
+# Handling Database Corruption
+
+* Status: accepted
+* Date: 2021-06-08
+
+## Context and Problem Statement
+
+Some of our users have corrupt SQLite databases and this makes the related
+component unusable.  The best way to deal with corrupt databases is to simply
+delete the database and start fresh (see #2628).  However, it's not so easy
+to know if a database operation error is from a corrupt database or from bad
+code.  We want a system that identifies and deals with database corruption but
```

Comment:
There are a few places below where you talk about "both" cases, so I'd like to suggest reframing this a little here - make the distinction between "permanent" and "transient" errors, and explicitly say we are considering issues in the code as transient because we can push a fix?